packages/material-ui/src/Typography/Typography.js

```diff
@@ -1,106 +1,69 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import capitalize from '../utils/capitalize';
+import typographyClasses, { getTypographyUtilityClass } from './typographyClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    margin: 0,
-  },
-  /* Styles applied to the root element if `variant="body2"`. */
-  body2: theme.typography.body2,
-  /* Styles applied to the root element if `variant="body1"`. */
-  body1: theme.typography.body1,
-  /* Styles applied to the root element if `variant="caption"`. */
-  caption: theme.typography.caption,
-  /* Styles applied to the root element if `variant="button"`. */
-  button: theme.typography.button,
-  /* Styles applied to the root element if `variant="h1"`. */
-  h1: theme.typography.h1,
-  /* Styles applied to the root element if `variant="h2"`. */
-  h2: theme.typography.h2,
-  /* Styles applied to the root element if `variant="h3"`. */
-  h3: theme.typography.h3,
-  /* Styles applied to the root element if `variant="h4"`. */
-  h4: theme.typography.h4,
-  /* Styles applied to the root element if `variant="h5"`. */
-  h5: theme.typography.h5,
-  /* Styles applied to the root element if `variant="h6"`. */
-  h6: theme.typography.h6,
-  /* Styles applied to the root element if `variant="subtitle1"`. */
-  subtitle1: theme.typography.subtitle1,
-  /* Styles applied to the root element if `variant="subtitle2"`. */
-  subtitle2: theme.typography.subtitle2,
-  /* Styles applied to the root element if `variant="overline"`. */
-  overline: theme.typography.overline,
-  /* Styles applied to the root element if `variant="inherit"`. */
-  inherit: {},
-  /* Styles applied to the root element if `align="left"`. */
-  alignLeft: {
-    textAlign: 'left',
-  },
-  /* Styles applied to the root element if `align="center"`. */
-  alignCenter: {
-    textAlign: 'center',
-  },
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
-    textAlign: 'right',
-  },
-  /* Styles applied to the root element if `align="justify"`. */
-  alignJustify: {
-    textAlign: 'justify',
-  },
-  /* Styles applied to the root element if `nowrap={true}`. */
-  noWrap: {
+const getTextColor = (color, palette) => {
+  if (color.indexOf('text') === 0) {
+    return palette.text[color.split('text').pop().toLowerCase()];
```

Comment:
In the future, we probably want to change the value supported by the color prop to match the format of https://github.com/mui-org/material-ui/blob/982abc5771c1ce2e0508d9bc79ad0bb9d43743d2/packages/material-ui-system/src/style.js#L5

```jsx
<Typograpy color="text.secondary" />
```

Or we could also say, drop the prop for the `sx` prop instead. It won't be consistent with the color prop of a button but that would be consistent with the other sx usages.

```jsx
<Typograpy sx={{ color: 'text.secondary' }} />
```

Reply:
Sounds good to me, let’s review the color prop on all components and see what patterns are there and make a proposal fo API. Sounds good?

Reply:
Yeah, it will be for a future iteration. It's outside of the scope of this effort.